src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -386,84 +404,233 @@ internal Database(C4Database* c4db)
+        /// </remarks>
+        /// <returns>
+        /// scope names of all existing scopes, in the order in which they were created.
+        /// </returns>
+        /// <exception cref="CouchbaseException">Thrown if an error condition is returned from LiteCore</exception>
+        /// <exception cref = "CouchbaseLiteException" > Thrown with <see cref="C4ErrorCode.NotFound"/>
+        /// if <see cref="Database"/> is closed</exception>
+        public IReadOnlyList<Scope> GetScopes()
         {
-            throw new NotImplementedException();
+            ThreadSafety.DoLocked(() =>
+            {
+                CheckOpen();
+                var arrScopes = Native.c4db_scopeNames(_c4db);
+                for (uint i = 0; i < Native.FLArray_Count((FLArray*)arrScopes); i++) {
+                    var scopeStr = (string)FLSliceExtensions.ToObject(Native.FLArray_Get((FLArray*)arrScopes, i));
+                    Scope s = null;
+                    if (!_scopes.ContainsKey(scopeStr)) {
+                        s = new Scope(this, scopeStr);
+                    } else {
+                        s = _scopes[scopeStr];
+                    }
+
+                    s.GetCollections();
+                    if (scopeStr == _defaultScopeName || s.Collections.Count > 0)
+                        _scopes.AddOrUpdate(scopeStr,
+                            (k) =>
+                            {
+                                return s;
+                            },
+                            (k, oldVal) =>
+                            {
+                                return oldVal;
+                            });
+                }
+
+                Native.FLValue_Release((FLValue*)arrScopes);
+            });
+
+            return Scopes;
         }
 
         /// <summary>
         /// Get a scope object by name. As the scope cannot exist by itself without having a collection, the nil 
         /// value will be returned if there are no collections under the given scope’s name.
-        /// Note: The default scope is exceptional, and it will always be returned.
         /// </summary>
-        /// <param name="name"></param>
-        /// <returns></returns>
-        public IScope GetScope(string name)
+        /// <remarks>
+        /// Note: The default scope is exceptional, and it will always be returned.
+        /// </remarks>
+        /// <param name="name">The name of the scope</param>
+        /// <returns>scope object with the given scope name</returns>
+        /// <exception cref="CouchbaseException">Thrown if an error condition is returned from LiteCore</exception>
+        /// <exception cref = "CouchbaseLiteException" > Thrown with <see cref="C4ErrorCode.NotFound"/>
+        /// if <see cref="Database"/> is closed</exception>
+        public Scope GetScope(string name)
         {
-            throw new NotImplementedException();
+            return _scopes.Values.SingleOrDefault(x => x.Name == name);
```

Comment:
This is going to return bad information if a scope is added via another Database instance.  I think if it is not in the cache, there needs to be a check against LiteCore.